consensus/istanbul/validator.go

```diff
@@ -85,16 +85,10 @@ type Validators []Validator
 // ----------------------------------------------------------------------------
 
 type ValidatorSet interface {
-	// Calculate the proposer
-	CalcProposer(lastProposer common.Address, round uint64)
-	// Get current proposer
-	GetProposer() Validator
-	// Check whether the validator with given address is the current proposer
-	IsProposer(address common.Address) bool
-	// Policy by which this selector chooses proposers
-	Policy() ProposerPolicy
 	// Sets the randomness for use in the proposer policy
 	SetRandomness(seed common.Hash)
+	// Sets the randomness for use in the proposer policy
```

Comment:
Is there a specific reason that `randomness` is left within the valset?    Should it be within round state?

Reply:
tbh, it was easier this way. `randomness` is set on `getOrderedValidators()` but not everytime we get a `ValidatorSet`. To change it, involved a larger refactor

One can always argue that valSet and randomness are constant within an epoch, and can be contained in the same object. But surely that object should not be called ValidatorSet.

Reply:
Can you include a comment mentioning that randomness is injected into the valset.

Also, I think your 2nd paragraph in the response makes a lot of sense and would be really helpful if you add a comment in the validatorset struct that states the same.

Reply:
done